mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostRegressionModelParitySpec.scala

```diff
@@ -36,6 +36,7 @@ class XGBoostRegressionModelParitySpec extends FunSpec
   private val xgboostParams: Map[String, Any] = Map(
     "eta" -> 0.3,
     "max_depth" -> 2,
+    "missing" -> 0.0f,
```

Comment:
Does this mean that missing are substituted with zeros ?

Reply:
I'm wondering, why not putting nulls here ?

Reply:
**Short Answer:**

1. This means that zeros will be assumed to be missing values.

2. We can't set it as anything other than 0.0 because xgboost throws the following error if we do that:

`java.lang.RuntimeException: you can only specify missing value as 0.0 (the currently set value NaN) when you have SparseVector or Empty vector as your feature format
`

**Detailed Answer:**

The logic behind not allowing any value other than 0.0  is that Sparks VectorAssembler always treats 0 as a missing value, so any SparseVector created by the Spark VectorAssembler will omit 0.0 treating it as missing. 
XGboost devs decided that to ensure compatibility with Sparks VectorAssembler they would enforce setting `missing` as 0.0 in their models whenever a SparseVector was used., otherwise; there would be incompatibility between DenseVector and SparseVector 

There is more info here: https://github.com/dmlc/xgboost/pull/4349/files